webapp/packages/core-sdk/src/Resource/CachedMapResource.test.ts

```diff
@@ -0,0 +1,253 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2023 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+import { CachedMapResource } from './CachedMapResource';
+import type { ResourceKey } from './ResourceKey';
+import { resourceKeyList } from './ResourceKeyList';
+
+interface ILoaderData {
+  id: string;
+  value: number;
+}
+
+const ERROR_ITEM_ID = 'error';
+
+const LOADER_DATA_MOCK: ILoaderData[] = [
+  {
+    id: '1',
+    value: 1,
+  },
+  {
+    id: '2',
+    value: 2,
+  },
+  {
+    id: ERROR_ITEM_ID,
+    value: 3,
+  },
+];
+
+const TEST_ERROR_MESSAGE = 'Test error';
+
+class TestMapResource extends CachedMapResource<string, ILoaderData> {
+  constructor() {
+    super();
+  }
+
+  async fetchAnError() {
+    return new Promise((_, reject) => {
+      setTimeout(() => {
+        reject(new Error(TEST_ERROR_MESSAGE));
+      }, 1);
+    });
+  }
+
+  private async fetchFromAPI(key: ResourceKey<string> | undefined): Promise<ILoaderData[]> {
+    return new Promise(resolve => {
+      setTimeout(() => {
+        if (key) {
+          const data = LOADER_DATA_MOCK.find(d => d.id === key);
+          if (data) {
+            resolve([data]);
+          }
+        } else {
+          resolve(LOADER_DATA_MOCK);
+        }
+      }, 1);
+    });
+  }
```

Comment:
Move this to `fetchMock` function outside of `TestMapResource`. This function must decide when to throw an exception